### ISSUE TITLE
Suggestion: Set mail server to defer delivery instead of reject on DNS Lookup failures

### ISSUE THREAD

**Issue by Targren** [eegeeZA/spamgourmet-clone#59]
(Ticket added at SysKoll's request)

With the semi-outage earlier this week being caused by a DNS server going screwy, I wanted to suggest setting up the mail server so that DNS issues return a 4xx error (Defer/temporary failure) error instead of the 550 error (Permanent failure). Most mail servers are configured not to keep retrying on permanent failure, so lots of emails were probably lost. 

`550-X-Host-Lookup-Failed: Reverse DNS lookup does not match for
10.0.0.1 **550** **(deferred)**`

The mail server might already be trying to do so, given the contradictory bounce message, so it could just be a misconfiguration.

**Comment by mikedlr** [comment]
I don't know what's going on there because 10.0.0.1 is a private address and so a bit weird to expect a DNS lookup to work on (unless that's a fake address for privacy reasons??) 

This requires serious thought and possibly redesign if done. Any form of increased delivery increases the resources that spammers may use and in particular, accepting mails that is likely to fail is something that can't really be done because we can't afford to send out bounce messages due to the [risk of backscatter](https://en.wikipedia.org/wiki/Backscatter_(email)). 

My idea for implementing this would be to keep a database record of the last successful delivery to a spamgourmet address and, if that's within a week, then accepting mails into a queue would likely not cause a large buildup. 

Even so, I'm not convinced that it isn't better to rely on retries from the originating server.

**Comment by Targren** [comment]
> unless that's a fake address for privacy reasons

Yes, exactly.  I sent the real server IP in the private DM on the BBS.

> accepting mails that is likely to fail is something that can't really be done because we can't afford to send out bounce messages due to the [risk of backscatter](https://en.wikipedia.org/wiki/Backscatter_(email)).

You don't have to worry about backscatter or sending out bounce messages. That's the sending MTA's responsibility. I'm just talking about changing the error code. The only difference that would make is that more than one delivery attempt would be made - a temporary outage shouldn't be a permanent failure. 

> Even so, I'm not convinced that it isn't better to rely on retries from the originating server.

That's exactly what this would facilitate. A 4xx error message tells the originating server to try again later. 5xx means "stroke off, that's not going to work".

**Comment by lwcorp** [comment]
I assume all of this relates to [0sg.net bouncing for bad reverse lookups?](https://bbs.spamgourmet.com/viewtopic.php?f=4&t=1901&sid=f02b6910280c9c7fe49202d780ad9c9d#p9069)

I'd therefore like to add one company that stopped sending to me because of it told me it's because of the `#FF0000` parts in [https://intodns.com/neverbox.com#:~:tex ... y%20parent](https://intodns.com/neverbox.com#:~:text=Missing%20nameservers%20reported%20by%20parent):

![image](https://github.com/spamgourmet/spamgourmet/assets/1773306/70f1ff6a-0031-40dd-a185-db3a1938fae6)
